src/utils/appInstaller/index.ts

```diff
@@ -0,0 +1,194 @@
+export default function AppInstaller(
+  installerOptions: AppInstallerOptions,
+  bre: BuidlerRuntimeEnvironment
+): AppInstaller {
+  /**
+   * @param name "finance"
+   */
+  return function appInstaller(
+    name: string,
+    appOptions?: AppOptions
+  ): Promise<AppInstalled> {
+    return _installExternalApp(
+      { name, ...(appOptions || {}), ...installerOptions },
+      bre
+    )
+  }
+}
+
+/* eslint-disable-next-line @typescript-eslint/explicit-function-return-type */
+async function _installExternalApp(
+  {
+    name,
+    version,
+    network = 'homestead',
+    initializeArgs,
+    skipInitialize,
+    apmAddress,
+    dao,
+    ipfsGateway
+  }: {
+    name: string
+    version?: string
+    initializeArgs?: any[]
+    skipInitialize?: boolean
+    network?: NetworkType
+    apmAddress: string
+    dao: KernelInstance
+    ipfsGateway: string
+  },
+  bre: BuidlerRuntimeEnvironment
+) {
+  const provider = new ethers.providers.InfuraProvider(network)
+  const etherscanProvider = new ethers.providers.EtherscanProvider(network)
+  const rootAccount: string = (await bre.web3.eth.getAccounts())[0]
+  const fullName = getFullName(name)
+
+  // Fetch version from external network
+  const versionData = await getExternalRepoVersion(fullName, version, provider)
+  const { contractAddress, contentURI } = versionData
+
+  // Fetch the deploy transaction from etherscan
+  const history = await etherscanProvider.getHistory(contractAddress, 0)
+  const deployTx = history[0]
+
+  const newDeployTx = await bre.web3.eth.sendTransaction({
+    from: rootAccount,
+    data: deployTx.data
+  })
+  const newContractAddress: string = newDeployTx.contractAddress
+
+  // Todo: construct valid deploy TX from the source code
+  // Make sure the contract code is correct before continuing
+  const codeNew = await bre.web3.eth.getCode(newContractAddress)
+  const codeReal = await provider.getCode(contractAddress)
+  if (codeNew !== codeReal)
+    throw Error('Error re-deploying contract code, it is not equal')
+
+  // Create new repo and publish its version
+  // Force the client to fetch from this specific ipfsGateway instead of localhost:8080
+  const shortName = name.split('.')[0]
+  const initialVersionArray = toVersionArray('1.0.0')
+  const contentUriHttpFromPublicGateway = utf8ToHex(
+    `http:${ipfsGateway}${getContentHash(contentURI)}`
+  )
+  const APMRegistry = bre.artifacts.require('APMRegistry')
+  const apmRegistry: APMRegistryInstance = await APMRegistry.at(apmAddress)
+  const repoAddress: string = await apmRegistry
+    .newRepoWithVersion(
+      shortName,
+      rootAccount,
+      initialVersionArray,
+      newContractAddress,
+      contentUriHttpFromPublicGateway
+    )
+    .then(txResponse => getLog(txResponse, 'NewRepo', 'repo'))
+  // Make sure the resulting repoAddress is accessible from the client
+  await assertEnsDomain(fullName, repoAddress)
```

Comment:
We may just want to do a sanity check on the above actions with the local chain in case the user installs the same app twice.